### PR TITLE
Add metrics-monitoring beats to resource monitoring

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?
 Closes https://github.com/elastic/elastic-agent/issues/4082

This PR is currently a draft, as the test borrows the integration setup code from https://github.com/elastic/elastic-agent/pull/4150. Once that PR is merged, we can refactor this to use the same code to setup + install integrations

## Why is it important?

This PR changes the behavior of the monitoring beats, so they _also_ monitor and report metrics on themselves. This fixes an issue where the CPU and memory usage that agent reports to fleet can be deceptive, as they don't include _all_ the beats that are running under agent.

I also did a light refactor of the monitoring setup so we use constants for the monitoring beats IDs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

### CODE REVIEW

**Comment by cmacknz** [comment]
Do the `beat/metrics` and `http/metrics` inputs report the resource usage and stats of the Beat they themselves are running in? Not just the other monitoring beats, as in `beat/metrics` is reporting the CPU usage of `beat/metrics-monitoring`, `http/metrics-monitoring` and `filestream-monitoring` and not just the latter two?

**Comment by fearful-symmetry** [comment]
> Do the `beat/metrics` and `http/metrics` inputs report the resource usage and stats of the Beat they themselves are running in? Not just the other monitoring beats, as in `beat/metrics` is reporting the CPU usage of `beat/metrics-monitoring`, `http/metrics-monitoring` and `filestream-monitoring` and not just the latter two?

That's the goal of this PR, yeah. If you want, you can verify the negative, and comment-out the lines that map in `v1_monitor.go`, and the resulting integration test will fail.

**Review by cmacknz — COMMENTED** [review]
How can we verify that the Fleet memory and CPU usage calculations are actually using the fields from the new documents?

If I install an agent built from this branch the CPU and memory in Fleet are exactly the same as one built from main. Possibly this is because they aren't using enough resources for their to be a detectable difference, but I would have expected memory to be higher at a minimum.

Screenshot 2024-03-08 at 3 59 37 PM

File: internal/pkg/agent/application/monitoring/v1_monitor_test.go [hunk 1]
```diff
@@ -186,11 +186,20 @@ func TestMonitoringConfigComponentFields(t *testing.T) {
...
+						t.Errorf("expecting fields['binary'] = 'filebeat', got %q", binary)
+					}
+					if componentID != "filestream-default" {
+						t.Errorf("expecting fields['id'] = 'filestream-default', got %q", componentID)
+					}
+				} else {
+					if binary != "filebeat" && binary != "metricbeat" {
+						t.Errorf("expected monitoring compoent to be metricbeat or filebeat, got %s", binary)
+					}
+					if componentID != "filestream-monitoring" && componentID != "beat/metrics-monitoring" && componentID != "http/metrics-monitoring" {
```
Comment (leehinman) [inline internal/pkg/agent/application/monitoring/v1_monitor_test.go:200]: optional.  I'm wondering if using the const `monitoringFilesUnitsID` might be better than the string "filestream-monitoring"

**Review by leehinman — APPROVED** [review]

File: testing/integration/metrics_monitoring_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		"filestream-monitoring",
+	}
+
+	require.Eventually(runner.T(), func() bool {
+		for _, cid := range componentIds {
+			query := genESQuery(agentStatus.Info.ID, cid)
+			res, err := estools.PerformQueryForRawQuery(ctx, query, "metrics-elastic_agent*", runner.info.ESClient)
+			require.NoError(runner.T(), err)
+			runner.T().Logf("Fetched metrics for %s, got %d hits", cid, res.Hits.Total.Value)
+			if res.Hits.Total.Value < 5 {
```
Comment (cmacknz) [inline testing/integration/metrics_monitoring_test.go:103]: Why `< 5`? Doesn't any amount of hits mean there was at least one document matching the query?

File: testing/integration/metrics_monitoring_test.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+						},
+					},
+					{
+						"match": map[string]interface{}{
+							"component.id": componentID,
+						},
+					},
+					{
+						"exists": map[string]interface{}{
+							"field": "system.process.cpu.total.value", // make sure we fetch documents that have the metric field used by fleet monitoring
```
Comment (cmacknz) [inline testing/integration/metrics_monitoring_test.go:129]: You could also check for the memory metric.

File: testing/integration/metrics_monitoring_test.go (hunk 2 above)
Comment (fearful-symmetry) [reply testing/integration/metrics_monitoring_test.go:103]: Yah, my concern was that we could end up in some freak accident where a test misconfiguration causes us to re-use an agent install, and thus an agent ID, leading to some overlap of results. Not sure if this is realistic, though.

Comment (cmacknz) [reply testing/integration/metrics_monitoring_test.go:103]: The agent ID you get on enrollment is unique within Fleet, so shouldn't be possible unless Fleet is completely broken.

This isn't protecting against that properly anyway IMO, it is just a magic number. 

A way that should actually work would be:
1. Install and enroll an agent with monitoring disabled.
2. Wait one full metrics collection cycle.
3. Ensure there are no hits in the metrics-* indices for that agent.
4. Turn monitoring on.
5. Ensure there are metrics now.

I'm not sure if all of that is actually worth it though.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**